crates/wash-lib/src/cli/mod.rs

```diff
@@ -345,6 +345,30 @@ pub fn labels_vec_to_hashmap(constraints: Vec<String>) -> Result<HashMap<String,
     Ok(hm)
 }
 
+/// This function is a simple helper to ensure that a component ID is a valid
+/// string containing only alphanumeric characters and underscores
+pub fn validate_component_id(id: &str) -> anyhow::Result<String> {
+    if id.chars().all(|c| c.is_alphanumeric() || c == '_') {
+        Ok(id.to_string())
+    } else {
+        bail!("Component ID must contain only alphanumeric characters and underscores")
+    }
+}
+
+/// This function is a simple helper to ensure that a component ID is a valid
+/// by transforming any non-alphanumeric characters to underscores
+pub fn sanitize_component_id(id: &str) -> String {
+    id.chars()
+        .map(|c| {
+            if c.is_alphanumeric() || c == '_' {
```

Comment:
This also can have dashes

Reply:
Super nit: consider making this char check a function so you can use it in the multiple places here